platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java

```diff
@@ -157,16 +171,57 @@ public DefaultInternalEventValidator(
      * @return true if the required fields of the event are non-null, otherwise false
      */
     private boolean areRequiredFieldsNonNull(@NonNull final PlatformEvent event) {
-        if (event.getSignature() == null) {
-            // do not log the event itself, since toString would throw a NullPointerException
-            nullUnhashedDataLogger.error(EXCEPTION.getMarker(), "Event has null signature");
-            nullUnhashedDataAccumulator.update(1);
+        final GossipEvent gossipEvent = event.getGossipEvent();
+        final EventCore eventCore = gossipEvent.eventCore();
+        if (eventCore == null
```

Comment:
Each of these `null` cases should have a unit test, to make sure it causes validation to fail

Reply:
added unit tests for each of them